src/system_info/system_info_utils.cc

```diff
@@ -48,7 +48,93 @@ char* GetDuidProperty() {
   }
   return s;
 }
-#endif
+
+#ifndef TIZEN_MOBILE
+GDBusConnection* GetDbusConnection() {
+  GError* error = NULL;
+  gchar* addr = g_dbus_address_get_for_bus_sync(G_BUS_TYPE_SYSTEM,
+                                                NULL, &error);
+  if (!addr) {
+    std::cout << "fail to get dbus addr: " << error->message << std::endl;
+    g_free(error);
+    return NULL;
+  }
+
+  GDBusConnection* bus_conn = g_dbus_connection_new_for_address_sync(addr,
+      (GDBusConnectionFlags)(G_DBUS_CONNECTION_FLAGS_AUTHENTICATION_CLIENT |
+      G_DBUS_CONNECTION_FLAGS_MESSAGE_BUS_CONNECTION), NULL, NULL, &error);
+  if (!bus_conn) {
+    std::cout << "fail to create dbus connection: "
+              << error->message << std::endl;
+    g_free(error);
+    return NULL;
+  }
+  return bus_conn;
+}
+
+char* OfonoGetModemPath(GDBusConnection* bus_conn) {
+  GError* error = NULL;
+  GVariant* modems_result = g_dbus_connection_call_sync(bus_conn,
+      kOfonoService, kOfonoManagerPath, kOfonoManagerIface,
+      "GetModems", NULL, NULL, G_DBUS_CALL_FLAGS_NONE, -1, NULL, &error);
+
+  if (!modems_result) {
+    std::cout << "ofono GetModems failed: " << error->message << std::endl;
+    g_error_free(error);
+    return NULL;
+  }
+
+  char* result_path = NULL;
+  GVariantIter* modems_iter;
+  g_variant_get(modems_result, "(a(oa{sv}))", &modems_iter);
+  char* path;
+  while (g_variant_iter_next(modems_iter, "(o@a{sv})", &path, NULL)) {
+    if (!path)
+      continue;
+
+    GVariant* modem_type_result = g_dbus_connection_call_sync(bus_conn,
+        kOfonoService, path, kOfonoModemIface, "GetProperties",
+        NULL, NULL, G_DBUS_CALL_FLAGS_NONE, -1, NULL, &error);
+
+    if (!modem_type_result)
+      continue;
+
+    GVariantIter* type_iter;
+    g_variant_get(modem_type_result, "(a{sv})", &type_iter);
+    GVariant* type_val;
+    char* key;
+    bool hardware_flag = false;
+    while (g_variant_iter_next(type_iter, "{sv}", &key, &type_val)) {
+      if (g_strcmp0(key, "Type") != 0)
+        continue;
+
+      const char* path_type = g_variant_get_string(type_val, NULL);
+      if (g_strcmp0(path_type, "hardware") == 0) {
+        result_path = path;
+        hardware_flag = true;
+        break;
+      } else if (g_strcmp0(path_type, "sap") == 0) {
+        result_path = path;
+        break;
+      }
+      break;
+    }
+
+    g_variant_unref(modem_type_result);
+    g_variant_iter_free(type_iter);
+    g_variant_unref(type_val);
+
+    if (hardware_flag)
+      break;
```

Comment:
I don't think it's better.
If so, every time ````g_strcmp0(key, "Type") != 0````, hardware_flag will be checked.
And after ````g_strcmp0(key, "Type") == 0````, even ````hardware_flag == true````, it will still run ````while (g_variant_iter_next(modems_iter, "(o@a{sv})", &path, NULL))````